File: documentapi/src/vespa/documentapi/messagebus/messages/wrongdistributionreply.cpp
// Copyright Vespa.ai. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.

#include "wrongdistributionreply.h"
#include <vespa/documentapi/messagebus/documentprotocol.h>

namespace documentapi {

WrongDistributionReply::WrongDistributionReply() :
    DocumentReply(DocumentProtocol::REPLY_WRONGDISTRIBUTION),
    _systemState()
{}

WrongDistributionReply::WrongDistributionReply(const string &systemState) :
    DocumentReply(DocumentProtocol::REPLY_WRONGDISTRIBUTION),
    _systemState(systemState)
{}

WrongDistributionReply::~WrongDistributionReply() {}

}

File: documentapi/src/main/java/com/yahoo/documentapi/messagebus/protocol/WrongDistributionReply.java
// Copyright Vespa.ai. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
package com.yahoo.documentapi.messagebus.protocol;

/**
 * @author Simon Thoresen Hult
 */
public class WrongDistributionReply extends DocumentReply {

    private String systemState;

    public WrongDistributionReply() {
        super(DocumentProtocol.REPLY_WRONGDISTRIBUTION);
    }

    public WrongDistributionReply(String systemState) {
        super(DocumentProtocol.REPLY_WRONGDISTRIBUTION);
        this.systemState = systemState;
    }

    public String getSystemState() {
        return systemState;
    }

    public void setSystemState(String state) {
        systemState = state;
    }
}
